src/platform/Zephyr/DiagnosticDataProviderImpl.cpp

```diff
@@ -74,5 +76,130 @@ CHIP_ERROR DiagnosticDataProviderImpl::GetCurrentHeapHighWatermark(uint64_t & cu
 #endif
 }
 
+CHIP_ERROR DiagnosticDataProviderImpl::GetRebootCount(uint16_t & rebootCount)
+{
+    uint32_t count = 0;
+    CHIP_ERROR err = ConfigurationMgr().GetRebootCount(count);
+
+    if (err == CHIP_NO_ERROR)
+    {
+        VerifyOrReturnError(count <= UINT16_MAX, CHIP_ERROR_INVALID_INTEGER_VALUE);
```

Comment:
Should be make it wrap around by doing  a `& 0xFFFF`? Thinking that once this hits the reboot loop will get stuck to invalid integer value for a very long time, I find that less useful than having some value that wraps. Thoughts?

Reply:
Actually I wasn't thinking about it much and just copied existing implementation from other platforms. I think that it can't wrap, as according to spec:

> The RebootCount attribute SHALL only be reset to 0 upon a factory reset of the Node

Maybe the best solution would be stopping at the max value and not incrementing it anymore, as:

> The RebootCount attribute SHALL indicate a best-effort count of the number of times the Node has
rebooted

Reply:
> A very large number (0xFFFF) seems better to me than saying 'error'. However this also breaks any checks of 'did this device ever reboot since last time'.

Yeah I agree. As you noted, the spec says:
> The RebootCount attribute SHALL indicate a best-effort count of the number of times the Node has
rebooted.

and also:
> The RebootCount attribute SHOULD be incremented each time the Node reboots.

But I believe shall is stronger statement than should, so providing max number instead of incrementing and ending with error seems to be best-effort solution and following the spec requirements. I decided to stay with the max value then.